File: TeamProject/2038L/2038L.cs
using System;

class Program
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        
        if (n == 1)
        {
            Console.WriteLine(2);
            return;
        }
        
        int result = n / 2;
        int remaining18 = n - (n / 2);
        
        if (n % 2 != 0)
        {
            result++;
            remaining18 -= 2;
        }
        
        if (remaining18 > 0)
        {
            result += remaining18 / 3;
            remaining18 %= 3;
        }
        
        result += (remaining18 + n + 1) / 2;
        
        Console.WriteLine(result);
    }
}

File: TeamProject/2038L/2038L.pl
calculate_boards(N, Result) :-
    (N =:= 1 -> Result = 2
    ;
    InitialBoards is N // 2,
    Remaining18 is N - (N // 2),
    
    (N mod 2 =\= 0 -> 
        Result1 is InitialBoards + 1,
        NewRemaining18 is Remaining18 - 2
    ;
        Result1 = InitialBoards,
        NewRemaining18 = Remaining18
    ),
    
    (NewRemaining18 > 0 ->
        AdditionalBoards is NewRemaining18 // 3,
        FinalRemaining18 is NewRemaining18 mod 3,
        Result2 is Result1 + AdditionalBoards
    ;
        Result2 = Result1,
        FinalRemaining18 = 0
    ),
    
    FinalCalculation is (FinalRemaining18 + N + 1) // 2,
    Result is Result2 + FinalCalculation
    ).

min_boards(N, Result) :-
    calculate_boards(N, Result).
